### PR TITLE
assets: Have `register_script()` take a textdomain

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

We're probably going to want to have Assets do something to interact
with the upcoming Webpack lazy-load i18n plugin when it registeres a
script using wp-i18n.

Even if that turns out not to be the case, this will still help make
sure that `wp_set_script_translations` gets called correctly when
needed.

#### Jetpack product discussion

PT: p9dueE-3MG-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Make sure stuff using `Assets::register_script()` still picks up translations, and no "doing it wrong" is triggered.

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
Looks good to me. 🚢

**Comment by anomiex** [comment]
Needs re-approval after merge conflict

**Review by jeherve — APPROVED** [review]

**Comment by anomiex** [comment]
Going to have to deploy this in two parts, first adding the `textdomain` to the calls, then after updating the Assets package we can remove the now-redundant `wp_set_script_translations()` calls.

**Comment by anomiex** [comment]
Part 1: r235416-wpcom

**Comment by anomiex** [comment]
Part 2: r235423-wpcom

File: projects/packages/assets/src/class-assets.php [hunk 1]
```diff
@@ -353,6 +355,15 @@ function ( $d ) {
...
 			self::instance()->add_async_script( $handle );
 		}
+		if ( $options['textdomain'] ) {
+			wp_set_script_translations( $handle, $options['textdomain'] );
+		} elseif ( in_array( 'wp-i18n', $options['dependencies'], true ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s is the script handle. */
+				esc_html( sprintf( __( 'Script "%s" depends on wp-i18n but does not specify "textdomain"', 'jetpack' ), $handle ) )
+			);
```
Comment (sergeymitr) [inline projects/packages/assets/src/class-assets.php:365]: Hi @anomiex.
The function seems to be missing the third argument, throws a fatal.

Comment (anomiex) [reply projects/packages/assets/src/class-assets.php:365]: Huh. I thought I had seen code elsewhere leaving that parameter out. Should have paid more attention.

#21852

**Pull request merged**